t/std/stdio_suite.c

```diff
@@ -0,0 +1,92 @@
+ *
+ * Copyright 2018, UT-Battelle, LLC.
+ *
+ * LLNL-CODE-741539
+ * All rights reserved.
+ *
+ * This is the license for UnifyCR.
+ * For details, see https://github.com/LLNL/UnifyCR.
+ * Please read https://github.com/LLNL/UnifyCR/LICENSE for full license text.
+ */
+
+#include <string.h>
+#include <mpi.h>
+#include <unifycr.h>
+#include "t/lib/tap.h"
+#include "t/lib/testutil.h"
+
+#include "stdio_suite.h"
+
+/* The test suite for sysio wrappers found in client/src/unifycr-stdio.c.
+ *
+ *
+ * To add new tests to existing sysio tests:
+ * 1. Add the tests (order matters) to the appropriate <stdio_function_name>.c
+ *    file.
+ * 2. Adjust the number (N) in `plan(N)` to account for the additional tests.
+ *
+ *
+ * When a new wrapper in unifycr-stdio.c needs to be tested:
+ * 1. Create a <stdio_function_name>.c file with a function called
+ *    <stdio_function_name>_test(char* unifycr_root) that contains all the TAP
+ *    tests specific to that wrapper.
+ * 2. Add the <stdio_function_name>_test to stdio_suite.h.
+ * 3. Add the <stdio_function_name>.c file to the /t/Makefile.am under the
+ *    appropriate test suite at the bottom.
+ * 4. The <stdio_function_name>_test function can now be called from this test
+ *    suite. */
+int main(int argc, char* argv[])
+{
+    int rank_num;
+    int rank;
+    char* unifycr_root;
+    int rc;
+
+    MPI_Init(&argc, &argv);
+    MPI_Comm_size(MPI_COMM_WORLD, &rank_num);
+    MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+
+    /* !!!IMPORTANT!!!
+     * Adjust this number when tests are added/removed from existing
+     * <stdio_function_name>.c files or when new test files are added.
+     *
+     * This determines how many tests should be run. Otherwise, in the event
+     * overall testing fails or bails out early, the framework will think only
+     * the tests that did run are all of the tests. This will result in the
+     * tests passing even when they didn't all run */
+    plan(8);
+
+    unifycr_root = testutil_get_mount_point();
+
+    /* Verify unifycr_mount succeeds. */
+    rc = unifycr_mount(unifycr_root, rank, rank_num, 0);
+    ok(rc == 0, "unifycr_mount at %s (rc=%d)", unifycr_root, rc);
+
+    if (rc != 0) {
+        BAIL_OUT("unifycr_mount in stdio_suite failed");
+    }
+
+    /* Add tests for new functions below in the order desired for testing.
+     *
+     * *** NOTE ***
+     * The order of tests does matter as some subsequent tests use functions
+     * that were already tested. Thus if an earlier test fails, it could cause
+     * later tests to fail as well. If this occurs, fix the bugs causing the
+     * tests that ran first to break as that is likely to cause subsequent
+     * failures to start passing.
+     *
+     * The number above each test is how many TAP tests are inside that
+     * function. They should add up to N-1 in plan(N) above */
+
+    /* 7 tests */
+    fopen_fclose_test(unifycr_root);
+
+    done_testing();
+
+    MPI_Finalize();
+
+    return 0;
+}
```

Comment:
@CamStan This is all really excellent and well documented. Is the plan(8) coming from the 7 tests in fopen_fclost_test + the one test in the main function in stdio_suite.c? Is there a way to have a counter for each test that is run so that we can pass the counter to the plan(<counter_variable>) to it instead of manually updating?

Reply:
@dsikich, yes, that is where the 8 is coming from. 

You make a great point; manually updating them really will be a pain. The tests are not run through any kind of iterative process that would make a counter automatic, but there still might be a better way to do it. Let me think though that. If not, it might be better to revert back to the `plan(NO_PLAN)` option than to have something we need to manually update.